ranger-gcs-connector-adapter/src/test/java/com/google/cloud/hadoop/ranger/gcs/connectorAdapter/RangerRequestTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ *
+ *  * Copyright 2020 Google Inc. All Rights Reserved.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *    http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.google.cloud.hadoop.ranger.gcs.connectorAdapter;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.hadoop.gcsio.authorization.ActionType;
+import com.google.cloud.hadoop.gcsio.authorization.StorageRequestSummary;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+@RunWith(JUnit4.class)
+public class RangerRequestTest {
+    private static final String USER = "test-user";
+    private static final String USERGROUPS = "grp1,grp2";
+
+    private static final String READ = "read";
+    private static final String WRITE = "write";
+
+    private static final String BUCKET = "bucket";
+    private static final String DIR = "/dir/path/";
+    private static final String OBJ = DIR + "obj";
+    private static final String ROOT = "/";
+
+    @Test
+    public void testListObject() throws IOException {
+        StorageRequestSummary.GcsStorage peer = new StorageRequestSummary.GcsStorage(BUCKET, OBJ, true);
+        StorageRequestSummary summary = new StorageRequestSummary(ActionType.LIST_OBJECT, Arrays.asList(peer));
+
+        List<RangerRequest> ret = RangerRequestFactory.createRangerRequest(summary, USER, USERGROUPS);
+
+        assertThat(ret.size() == 1).isTrue();
+        assertThat(ret.get(0).getUser()).isEqualTo(USER);
```

Comment:
you can extract ret from the list so you do not have to do ret.get(0) every time. Same for all the other tests.